x-pack/plugins/lists/server/routes/list_index/read_list_index_route.ts

```diff
@@ -30,32 +28,32 @@ export const readListIndexRoute = (router: ListsPluginRouter): void => {
 
       try {
         const lists = await getListClient(context);
-        const listIndexExists = await lists.getListIndexExists();
-        const listItemIndexExists = await lists.getListItemIndexExists();
+        const listDataStreamExists = await lists.getListDataStreamExists();
+        const listItemDataStreamExists = await lists.getListItemDataStreamExists();
 
-        if (listIndexExists || listItemIndexExists) {
+        if (listDataStreamExists || listItemDataStreamExists) {
```

Comment:
These `IF` statements are not going to work as initially intended.

First `if (listDataStreamExists || listItemDataStreamExists) {` will handle three out of four possible cases:
> 1. listDataStreamExists = true && listItemDataStreamExists = true
> 2. listDataStreamExists = false && listItemDataStreamExists = true
> 3. listDataStreamExists = true && listItemDataStreamExists = false

Then, second `if (!listDataStreamExists && listItemDataStreamExists) {` statement will handle the last possible case:
> 4. listDataStreamExists = false && listItemDataStreamExists = false

This means, that last two statements (`if (!listItemDataStreamExists && listDataStreamExists) {` and final `else` case) will never be called.